gamemodes/horde/gamemode/sv_rank.lua

```diff
@@ -142,28 +142,28 @@ function HORDE:LoadRank(ply)
 end
 
 local ExpMultiConvar = GetConVar("horde_experience_multiplier")
+local StartXPMult = Horde.Difficulty[Horde.CurrentDifficulty].xpMultiStart
+local EndXPMult = Horde.Difficulty[Horde.CurrentDifficulty].xpMultiEnd
 
 if GetConVar("horde_enable_sandbox"):GetInt() == 0 and GetConVar("horde_enable_rank"):GetInt() == 1 then
 	hook.Add("Horde_OnEnemyKilled", "Horde_GiveExp", function(victim, killer, wpn)
-
-		local ExpMulti = ExpMultiConvar:GetInt()
 			
-		if HORDE.CurrentDifficulty >= 5 and HORDE.current_wave >= 4 then
-			ExpMulti = 2 * ExpMulti
-		end
+		local WavePercent = Horde.current_wave / 10 --If we ever change the max # of waves this will have to become dynamic instead of 10
+		local RoundXPMulti = StartXPMult+(WavePercent*(EndXPMult-StartXPMult)) --This gets the xp multi number between min and max multi based on round
```

Comment:
`EndXPMult-StartXPMult` can be calculated outside the hook

Reply:
added, also added camelcase but already resolved it

Reply:
added